src/mappings/prices.ts

```diff
@@ -16,37 +17,64 @@ const KrakenKsmUsd = readCSV(
 const KrakenXlmUsd = readCSV(
     `ohlcv/XLMUSD_${TIMEFRAME_INTERVAL_IN_MINUTES}.csv`
 )
+const KrakenEurUsd = readCSV(
+    `ohlcv/EURUSD_${TIMEFRAME_INTERVAL_IN_MINUTES}.csv`
+)
+const KrakenAudUsd = readCSV(
+    `ohlcv/AUDUSD_${TIMEFRAME_INTERVAL_IN_MINUTES}.csv`
+)
 // We don't have the DOT price available on Pendulum because the DIA pallet was not configured yet
 // const KrakenDotUsd = readCSV(
 //     `ohlcv/DOTUSD_${TIMEFRAME_INTERVAL_IN_MINUTES}.csv`
 // )
 
-export const KsmRollingAverage = new RollingAverage('KSM-USD', 0.01)
-export const XlmRollingAverage = new RollingAverage('XLM-USD', 0.01)
+interface PriceDeviation {
+    timestamp: number
+    deviation: number | null
+}
 
-// export const DotRollingAverage = new RollingAverage(0.01)
+export const deviationObject: { [key: string]: PriceDeviation[] } = {}
 
 async function handleAnalysisFor(
     ctx: EventHandlerContext,
     coinInfo: any,
     dataSet: Promise<OHLCV[]>,
-    rollingAverage: RollingAverage
+    symbol: string
 ) {
     // We try to pick the timestamp reported by the block, because the timestamp in the coin info might not be up-to-date
     const timestamp =
         ctx.block.timestamp || Number(coinInfo.lastUpdateTimestamp)
-
     try {
         const data = await dataSet
         const ohlcvAtTime = getOHLCVAtTime(data, timestamp)
-        const krakenPrice = ohlcvAtTime.close
-        let diaPrice = Number(coinInfo.price.toString())
-        // We downscale the price by 12 decimals
-        diaPrice = diaPrice / 10 ** 12
-        const priceDiff = Math.abs(krakenPrice - diaPrice)
-        // We calculate the percentage difference between the two prices
-        const priceDiffPercentage = priceDiff / krakenPrice
-        rollingAverage.add(priceDiffPercentage)
+
+        if (!ohlcvAtTime) {
+            console.log('No OHLCV data for', symbol, 'at', timestamp)
+            let deviation: PriceDeviation = {
+                timestamp,
+                deviation: null,
+            }
+            saveDeviation(deviation, symbol)
+        }
+
+        let diaPrice = Number(coinInfo.price.toString()) / 10 ** 12
+        if (diaPrice < ohlcvAtTime!.low) {
+            const price_deviation =
+                ((ohlcvAtTime!.low - diaPrice) / ohlcvAtTime!.low) * 100
+            let deviation: PriceDeviation = {
+                timestamp,
+                deviation: price_deviation,
+            }
+            saveDeviation(deviation, symbol)
+        } else if (diaPrice > ohlcvAtTime!.high) {
+            const price_deviation =
+                ((diaPrice - ohlcvAtTime!.high) / ohlcvAtTime!.high) * 100
```

Comment:
I think it's more accurate to divide `(a - b) / a` like we did in l.63 
```suggestion
                ((diaPrice - ohlcvAtTime!.high) / diaPrice) * 100
```

Reply:
You are right! Missed that.

Reply:
I now think I prefer the other way around, because I think we may want the percentage above (or below) the "true" value or limit, which is `ohlcvAtTime!.high`. What do you think @ebma ?

Reply:
Ah hmm, well you have a point. I'm fine if we keep it like this then 👍